consensus/replay.go

```diff
@@ -252,15 +252,10 @@ func (h *Handshaker) Handshake(proxyApp proxy.AppConns) (string, error) {
 	}
 	appHash := res.LastBlockAppHash
 
-	h.logger.Info("ABCI Handshake App Info",
-		"height", blockHeight,
-		"hash", appHash,
-		"software-version", res.Version,
-		"protocol-version", res.AppVersion,
-	)
-
-	// Only set the version if there is no existing state.
-	if h.initialState.LastBlockHeight == 0 {
```

Comment:
[Question] We are incorporating extra conditions into this if statement; was this deliberate? From my perspective, it seems that the original version accomplished the same effect as the new modifications, except that the recent changes introduce more limitations.

Reply:
Yes we want to add more limitations as without these added constraints we get the app version returned from Info always overwriting the app version derived from the genesis file but that shouldn't always be the case

Reply:
Thanks for clarification!